torchdata/datapipes/iter/util/rar_archive_loader.py

```diff
@@ -0,0 +1,73 @@
+import io
+import os.path
+from unittest.mock import patch
+from typing import Tuple, Iterator
+
+from torchdata.datapipes.utils import StreamWrapper
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils.common import validate_pathname_binary_tuple
+
+
+class RarfilePatcher:
+    def __init__(self):
+        from rarfile import DirectReader
+
+        unpatched_read = DirectReader._read
+
+        def patched_read(self, cnt):
+            whence = self._cur.add_size - self._cur_avail
+            self._fd.seek(self._inf.header_offset, self._cur.add_size - self._cur_avail)
```

Comment:
I did something stupid. Could you please modify it to `self._fd.seek(self._inf.header_offset, whence)`?